modules/user/src/main/PasswordHasher.scala

```diff
@@ -0,0 +1,55 @@
+package lila.user
+
+import javax.crypto.Cipher
+import javax.crypto.spec.{ IvParameterSpec, SecretKeySpec }
+import java.util.Base64
+
+import com.roundeights.hasher.Implicits._
+
+/**
+ * Encryption for bcrypt hashes.
+ *
+ *  The AES parameters were chosen based on application
+ *  (bcrypt hashes) and are incorrect for general use.
+ *
+ *  In particular, CTS reveals input length and a
+ *  static IV usually compromises confidentiality.
```

Comment:
So looking at this again ... It certainly helps that the plaintexts are probably indistinguishable from random, but we still should not do this. The IV is supposed to be random (that also means independent of the plaintext!).

Alternatives:

* Use proper random IVs.
* Use the salt (which is indeed random), but keep it out of the plaintext.

Reply:
I think it's fine, AES is defense in depth anyway. we don't need an IV

Reply:
I can see what you're saying but leaving the salt outside aes leaks info about the state too, right? (albeit less since you'd need password)

Reply:
I don't want to use proper IVs because that increases storage

Reply:
fixed, I think